File: windows/src/GoalMaker.Core/Planning/ProjectItem.cs
namespace GoalMaker.Core.Planning;

/// <summary>
/// A project as the screens and rules see it (docs/projects.md): where its code lives, which area it
/// belongs to, and whether it is active, paused or done.
/// </summary>
public sealed record ProjectItem(string Id, string Name)
{
    public string Description { get; init; } = string.Empty;

    public string? AreaId { get; init; }

    public string Status { get; init; } = ProjectRules.Active;

    public string? RepositoryUrl { get; init; }

    public string? LocalFolder { get; init; }

    public string Notes { get; init; } = string.Empty;

    public double Position { get; init; }

    public bool Deleted { get; init; }
}

File: android/app/src/main/kotlin/com/goalmaker/app/application/planning/ProjectItem.kt
package com.goalmaker.app.application.planning

/**
 * A project as the screens and rules see it (docs/projects.md): where its code lives, which area it
 * belongs to, and whether it is active, paused or done.
 */
data class ProjectItem(
    val id: String,
    val name: String,
    val description: String = "",
    val areaId: String? = null,
    val status: String = ProjectRules.ACTIVE,
    val repositoryUrl: String? = null,
    val localFolder: String? = null,
    val notes: String = "",
    val position: Double = 0.0,
    val deleted: Boolean = false,
)
